ajax/impact.php

```diff
@@ -52,25 +52,27 @@
 
       // Check required params
       if (empty($itemtype) || empty($items_id)) {
-         Toolbox::throwBadRequest("Missing itemtype or items_id");
+         Toolbox::throwError(400, "Missing itemtype or items_id");
       }
 
       // Check that the the target asset exist
       if (!Impact::assetExist($itemtype, $items_id)) {
-         Toolbox::throwBadRequest("Object[class=$itemtype, id=$items_id] doesn't exist");
+         Toolbox::throwError(400, "Object[class=$itemtype, id=$items_id] doesn't exist");
       }
 
       // Prepare graph
       $item = new $itemtype;
       $item->getFromDB($items_id);
       $graph = Impact::makeDataForCytoscape(Impact::buildGraph($item));
       $params = Impact::prepareParams($item);
+      $readonly = !Session::haveRight(get_class($item)::$rightname, UPDATE);
```

Comment:
Maybe this would be more accurate.
```suggestion
      $readonly = !$item->can($items_id, UPDATE);
```